### PR TITLE
Update foojay plugin version to 0.7.0

### PR DESCRIPTION
**Pull request by alllex**
A new version `0.7.0` of the Foojay Toolchains plugin has been released, but we still use and mention in the docs a pretty old one -- `0.4.0`.

### CODE REVIEW

**Review by lkasso — APPROVED** [review]

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -65,6 +65,7 @@
       </trusted-artifacts>
       <ignored-keys>
          <ignored-key id="74dafdfd6dae2441" reason="Key couldn't be downloaded from any key server"/>
+         <ignored-key id="e2f38302c8075e3d" reason="Key couldn't be downloaded from any key server"/>
```
Comment (ljacomet) [inline gradle/verification-metadata.xml:68]: ❌ This needs to be removed, it is an artifact from the key failing to download.

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -1345,6 +1346,11 @@
             <sha256 value="bdce53a751fdb27af6608039df81214ba22d902ed4169540a3daeb5828c99cad" reason="Artifact is not signed"/>
          </artifact>
       </component>
+      <component group="org.gradle.toolchains" name="foojay-resolver" version="0.7.0">
```
Comment (ljacomet) [inline gradle/verification-metadata.xml:1349]: ❌ This should not appear as we trust the key itself.

**Review by ljacomet — CHANGES_REQUESTED** [review]
See remarks below, there should be no changes to dependency verification for this update.

File: gradle/verification-metadata.xml (hunk 2 above)
Comment (ljacomet) [reply gradle/verification-metadata.xml:1349]: But the trust of the key needs to be added. #26194 is taking care of this

**Comment by alllex** [comment]
@bot-gradle test and merge

**Pull request merged**